include/fmt/format.h

```diff
@@ -1885,6 +1891,13 @@ class arg_formatter_base {
 
   template <typename T, FMT_ENABLE_IF(std::is_floating_point<T>::value)>
   iterator operator()(T value) {
+    if (const_check(
+            (std::is_same<T, float>::value && !FMT_USE_FLOAT) ||
+            (std::is_same<T, double>::value && !FMT_USE_DOUBLE) ||
+            (std::is_same<T, long double>::value && !FMT_USE_LONG_DOUBLE))) {
```

Comment:
This can be factored out into a constexpr function and reused here and above.

Reply:
OK, is the latest commit sort of what you had in mind?

Reply:
Yes, thank you.